File: tools/keys/karabiner/lib/conditions.ts
// deno-lint-ignore-file camelcase
// from https://github.com/denolfe/dotfiles/blob/master/karabiner/lib/conditions.ts
import { Condition } from "https://deno.land/x/karabiner@v0.2.0/karabiner.ts";

type FrontmostApp = 'chrome' | 'vscode' | 'slack' | 'safari' | 'brave'

const bundleMap: Record<FrontmostApp, string> = {
  chrome: 'com.google.Chrome',
  vscode: 'com.microsoft.VSCode',
  slack: 'com.tinyspeck.slackmacgap',
  safari: 'com.apple.Safari',
  brave: 'com.brave.Browser'
}

export function ifApp(app: FrontmostApp): Condition {
  return {
    type: 'frontmost_application_if',
    bundle_identifiers: [bundleMap[app]],
  }
}

export function notApp(app: FrontmostApp): Condition {
  return {
    type: 'frontmost_application_unless',
    bundle_identifiers: [bundleMap[app]],
  }
}

export const nonAppleDevice: Condition = {
  type: 'device_unless',
  identifiers: [
    {
      vendor_id: 1452,
    },
  ],
}

File: tools/keys/karabiner/conditions.py
IF_DEVICE_IS_APPLE_INTERNAL_KEYBOARD = [
    {
        "type": "device_if",
        "identifiers": [
            {
                "vendor_id": 1452,
                "product_id": 832,
                "is_keyboard": True,
            }
        ],
    },
]

IF_DEVICE_IS_EXTERNAL_MOUSE = [
    {
        "type": "device_if",
        "identifiers": [
            {
                "vendor_id": 1111,
                "product_id": 111,
            }
        ],
    }
]

IF_FRONT_APPLICATION_IS_BRAVE = [
    {
        "type": "frontmost_application_if",
        "bundle_identifiers": [
            r"^com\.brave\.Browser$",
        ],
    }
]
